dev_guide/getting_traffic_into_cluster.adoc

```diff
@@ -0,0 +1,112 @@
+{product-author}
+{product-version]
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+ifdef::openshift-origin,openshift-enterprise,openshift-dedicated[]
+== Overview
+There are many ways to access the cluster. This section describes some
+commonly used approaches.
+
+The recommendation is:
+
+- When possible use the Router,
+- otherwise for TCP/UDP configure the service to provide access.
+
+The admin configures DNS records to point to IP addresses in the cluster
+that are exposed outside the cluster.  These addresses are commonly the
+IP addresses of nodes, Virtual IP addresses (VRRP) or Service NodePort
+addreses, Service externalIPs, Service ingressIP, etc..  The admin must
+make sure the routing tables permit the request to reach the IP address
+and iptables accept the packets as well.
+
+The xref:../admin_guide/high_availability.adoc#[High Availability] section
+describes how to make this highly available using replicated services.
+
+- xref:../dev_guide/getting_traffic_into_cluster.adoc#using-a-router[Using a router] to serve HTTP/HTTPS(SNI)/TLS (SNI)
+- xref:../dev_guide/getting_traffic_into_cluster.adoc#using-tcp[Service NodePort] and VIP
+- xref:../dev_guide/getting_traffic_into_cluster.adoc#pod-network-to-outside[Access pod network with a load balancer]
+
+You can also use the wildcard DNS for the router + the nodeport
+
+endif::[]
+
+
+[[using-a-router]]
+== Using a Router
+
+This is the most common way to access the cluster. A router is configured
+to accept external reuqests and proxy them based on the configured routes.
+This is limited to HTTP/HTTPS(SNI)/TLS(SNI) which covers web applications.
+
+Setup router nodes (either TCP load balance or ipfailover VRRP setup) in
+an HA config.  When asking for nodeports, use the router nodes with their
+VIPs (VRRP) or with a range TCP load balancer (cloud, F5, ingerssIP) as
+the destination IP.
+
+Since the router listens on ports 80 and 443 the number of routers is
+limited to the number of nodes that have ports 80 and 443 available.
+
+The router IP can be the node's IP or a VRRP provided VIP
+(xref:../admin_guide/high_availability.adoc#[virtual IP]).
+
+A set of routes can be created in the various projects. The overall
+set of routes is available to the set of routers. Each router selects
+from the set of routes.
+
+The admin configures DNS "A" records to point a set of paths to the
+desired router IP address.  The admin may have to change the routing
+table and iptable to make sure packets reach the routers.
+
+
+The xref:../admin_guide/high_availability.adoc#[High Availability]
+section describes how to configure a router for high availbility service
+using multiple replicas.
+
+
+[[using-tcp]]
+== Using Service NodePort
+
+This approach exposes TCP/UDP streams in pods to external hosts.
+
+The user confiugres the service to expose the needed ports and ipaddresses.
+
+The user can assign a service.port.nodePort and service spec.externalIPs to
```

Comment:
ExternalIP needs its own section.  http://kubernetes.io/docs/user-guide/services/#external-ips

The external IP exposes the _service port_ (not the nodeport) on the ip address given.  If a nodeport is specified too, I think you get the nodeport port bound on all nodes AND you get the service port exposed on the external IP address (they may be different ports).